### PR TITLE
[4.2.0] Update the minimum recommended core count

### PR DESCRIPTION
**Pull request by DinithiDiaz**
## Purpose
This PR is to update the minimum recommended core count.

This will fix:
- https://github.com/wso2-enterprise/wso2-mi-internal/issues/112
- https://github.com/wso2-enterprise/wso2-mi-internal/issues/770

### CODE REVIEW

File: en/docs/install-and-setup/install/installation-prerequisites.md [hunk 1]
```diff
@@ -7,36 +7,40 @@ Prior to installing WSO2 Micro Integrator, make sure that the appropriate prereq
 <table>
   <tr>
     <td>
-      <b>Docker</b>
+      <b>Monolithic deployment</b>
```
Comment (isudana) [inline en/docs/install-and-setup/install/installation-prerequisites.md:10]: Monolithic -> Centralized (ESB Style) Deployment

File: en/docs/install-and-setup/install/installation-prerequisites.md [hunk 2]
```diff
@@ -7,36 +7,40 @@ Prior to installing WSO2 Micro Integrator, make sure that the appropriate prereq
...
-          Minimum 0.5 core per Micro Integrator Docker instance.
+          <code>~512</code> MB heap size. This is generally sufficient for processing smaller messages. However, the requirements vary with larger message sizes and the number of messages processed concurrently.
         </li>
       </ul>
     </td>
   </tr>
   <tr>
     <td>
-      <b>Virtual Machine (VM)/Physical</b>
+      <b>Microservices deployment</b>
```
Comment (isudana) [inline en/docs/install-and-setup/install/installation-prerequisites.md:28]: Decentralized (Microservices Style) Deployment

File: en/docs/install-and-setup/install/installation-prerequisites.md (hunk 1 above)
Comment (DinithiDiaz) [inline en/docs/install-and-setup/install/installation-prerequisites.md:10]:
```suggestion
      <b>Centralized (ESB Style) Deployment</b>
```

File: en/docs/install-and-setup/install/installation-prerequisites.md [hunk 3]
```diff
@@ -7,36 +7,40 @@ Prior to installing WSO2 Micro Integrator, make sure that the appropriate prereq
...
```
Comment (DinithiDiaz) [inline en/docs/install-and-setup/install/installation-prerequisites.md:28]:
```suggestion
      <b>Decentralized (Microservices Style) Deployment</b>
```

File: en/docs/install-and-setup/install/installation-prerequisites.md (hunk 2 above)
Comment (DinithiDiaz) [reply en/docs/install-and-setup/install/installation-prerequisites.md]: Fixed via 5be28bf6ac85209df46d354a8c57f16793438003.

**Review by arunans23 — APPROVED** [review]

**Pull request merged**